### ISSUE TITLE
Add class when product added to cart

### ISSUE THREAD

**Issue by thagxt** [EvgeniyMukhamedjanov/liquid-ajax-cart#86]
## Summary

Add an additional class that gets added when a product is in the process of being added to the cart. 

## Current Behavior

The `.js-ajax-cart-processing` class is currently added whenever the page goes out of focus and back into focus. This makes it unreliable for detecting when a specific product is being added to the cart.

## Expected Behavior

- Add a class like `.js-ajax-cart-adding-to-cart` that gets added to the page when a product add to cart request begins. 
- Remove `.js-ajax-cart-adding-to-cart` when the add to cart request finishes.

## Reason

We need a reliable way to detect when an individual product is in the process of being added to the cart. This will allow things like:

- Showing a loading indicator specifically for the add to cart action
- Temporarily disabling the add to cart button 
- Communicating to the user that the requested product is being added
- Communicating to the user that the product has been added.

Without a dedicated class for add to cart in progress, these types of UX enhancements are difficult.

Let me know if any clarification or additional details are needed!

**Comment by EvgeniyMukhamedjanov** [comment]
Hi @thagxt .
The "loading" attribute doesn't work for you?
https://liquid-ajax-cart.js.org/v2/product-forms/#loading-state

**Comment by thagxt** [comment]
Sure, the button works fine when it's inside the `<ajax-cart-product-form>`. 

But in my case: I've a "loader indicator" in the header of the website, and since it is outside `<ajax-cart-product-form>` it's not possible to apply styles to it like in the example. I use this loader to visually inform the user that page is still being loaded, when additional elements are loaded via AJAX and now I want it to show up when someone adds a product to the cart too.

I know I can do it via JS, however it would be handy to have a class that's being added from the script directly.

**Comment by EvgeniyMukhamedjanov** [comment]
I see.
It seems it is more advanced and custom case, because in the most cases we wanna show a global loading indicator for all the Ajax requests (not just for the add to cart request) which is how the `.js-ajax-cart-processing` works: it appears when a user is adding/removing/changing/updating items in the cart.

So I would prefer to solve this case by extending the library using custom JavaScript and Liquid Ajax Cart API rather than including the code to the library.

I'm sure you know, but just in case I'm going to leave the code here how to add/remove CSS class on adding a product:
```javascript
document.addEventListener('liquid-ajax-cart:request-start', event => {
  const {requestState} = event.detail;
  if (requestState.requestType === 'add') {
    document.body.classList.add('js-ajax-cart-adding-to-cart');
  }
});

document.addEventListener('liquid-ajax-cart:request-end', event => {
  const {requestState} = event.detail;
  if (requestState.requestType === 'add') {
    document.body.classList.remove('js-ajax-cart-adding-to-cart');
  }
});
```

Btw if the only thing that doesn't fit your needs of the `.js-ajax-cart-processing` is that it is added when page gets/loses focus, you can switch off updating the cart on page focus:
```javascript
window.liquidAjaxCart.conf("updateOnWindowFocus", false);
```

Let me know if I didn't get your question correctly.

**Comment by thagxt** [comment]
Thanks for your prompt response, as always. 

I appreciate your thoughts on the matter and understand where you're coming from; the current approach, as you mentioned, makes sense for most use cases; however, having a global class added to the HTML tag for a fundamental action such as the "add to cart" one can be useful in cases like mine or others where you need to control things outside the currently observed regions, in my opinion. 

Anyway, I recognize that you may not want to bloat the script because it is easily attainable with a few lines of JS and can be left optional.

**Comment by EvgeniyMukhamedjanov** [comment]
Yeah, I would prefer to cover such cases via using Liquid Ajax Cart JS API - that is why I provided it 🙂
Otherwise I would have to add classes for each action - add to cart, remove, update, change etc. to be consistent, but they won't be used in most cases.
